File: spring-boot/src/main/java/springboot/exceptions/ErrorCode.java
package springboot.exceptions;

import lombok.Getter;

public enum ErrorCode {
    BAD_REQUEST(400, "9000", "Bad request."),
    UNAUTHORIZED(401, "9001", "Unauthorized."),
    NOT_FOUND(404, "9002", "Not found"),
    INTERNAL_SERVER_ERROR(500, "9003", "Internal server error");

    @Getter private final int httpCode;
    @Getter private final String code;
    @Getter private final String message;

    private ErrorCode(int httpCode, String code, String message) {
        this.httpCode = httpCode;
        this.code = code;
        this.message = message;
    }
}

File: spring-boot-kt/src/main/kotlin/real/world/springbootkt/global/common/ErrorCode.kt
package real.world.springbootkt.global.common

import com.fasterxml.jackson.annotation.JsonValue

enum class ErrorCode(
    val httpCode: Int,
    @JsonValue val code: String,
    val message: String
) {
    BAD_REQUEST(400, "9000", "Bad request."),
    UNAUTHORIZED(401, "9001", "Unauthorized."),
    FORBIDDEN(403, "9002", "Forbidden."),
    NOT_FOUND(404, "9002", "Not found"),
    INTERNAL_SERVER_ERROR(500, "9003", "Internal server error");
}
